src/uwtools/resources/jsonschema/esg-grid.jsonschema

```diff
@@ -0,0 +1,119 @@
+{
+  "allOf": [
+    {
+      "if": {
+        "not": {
+          "required": [
+            "namelist.base_file"
+          ]
+        }
+      },
+      "then": {
+        "properties": {
+          "namelist": {
+            "properties": {
+              "update_values": {
+                "required": [
+                  "delx",
+                  "dely",
+                  "lx",
+                  "ly",
+                  "pazi",
+                  "plat",
+                  "plon"
+                ]
+              }
+            },
+            "required": [
+              "update_values"
+            ]
+          }
+        }
+      }
+    }
+  ],
+  "properties": {
+    "esg_grid": {
+      "additionalProperties": false,
+      "properties": {
+        "execution": {
+          "$ref": "urn:uwtools:execution-serial"
+        },
+        "namelist": {
+          "additionalProperties": false,
+          "anyOf": [
+            {
+              "required": [
+                "base_file"
+              ]
+            },
+            {
+              "required": [
+                "update_values"
+              ]
+            }
+          ],
```

Comment:
I see now that it *is* ok for neither `base_file` or `update_values` to be provided: Just don't provide the `namelist` block at all! But if it _is_ provided, then at least one of `base_file` or `update_values` is required. That makes sense, and means that users can place their own namelist file in the run directory and not mention `namelist` at all in their UW YAML. Sorry for the confusion.

Reply:
It's not that `base_file` is optional, it's that all of `namelist` is optional. We've previously discussed leaving open the possibility that a user might place their own namelist file in a run directory. Of course, they could also automate that by using `base_file`, but they might do it completely outside the driver. It looks like this is the choice we've made in all the driver schemas so far: `namelist` is optional, but if you provide it, you have to provide at least one of `base_file` and/or `update_values`. If `namelist` isn't provided, and if the driver doesn't find it in the run directory already, we'll get `WARNING` messages about the unfulfilled requirement. (Well, I see that in `chgres_cube` we `["namelist"]` instead of `.get("namelist")`, so we should fix that and check the other drivers for agreement between the schema and the `dict` access method used for optional items.)

So... I think we should decide what we really want and then double check that that's the case. I'm happy to do the check-and-fix part, but a design decision has to come first.